XamarinCommunityToolkit/Effects/SafeAreaPaddingEffectRouter.iOS.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using UIKit;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+using XamarinCommunityToolkit.Effects;
+using Effects = XamarinCommunityToolkit.iOS.Effects;
+
+[assembly: ResolutionGroupName(nameof(XamarinCommunityToolkit))]
+[assembly: ExportEffect(typeof(Effects.SafeAreaPaddingEffectRouter), nameof(Effects.SafeAreaPaddingEffectRouter))]
+namespace XamarinCommunityToolkit.iOS.Effects
+{
+    public class SafeAreaPaddingEffectRouter : PlatformEffect
+    {
+        Thickness initialPadding;
+
+        protected override void OnAttached()
+        {
+            if (!(Element is Layout element))
+                return;
+
+            if (UIDevice.CurrentDevice.CheckSystemVersion(11, 0))
+            {
+                var insets = UIApplication.SharedApplication.Windows[0].SafeAreaInsets;
+
+                initialPadding = element.Padding;
+
+                if (insets.Top > 0)
+                {
+                    var safeArea = SafeAreaPaddingEffect.GetSafeAreaPadding(element);
+
+                    element.Padding = new Thickness(
+                        initialPadding.Left + (safeArea.Left ? insets.Left : 0),
+                        initialPadding.Top + (safeArea.Top ? insets.Top : 0),
+                        initialPadding.Right + (safeArea.Right ? insets.Right : 0),
+                        initialPadding.Bottom + (safeArea.Bottom ? insets.Bottom : 0)
+                    );
+
+                    return;
+                }
+            }
+
+            element.Padding = initialPadding;
```

Comment:
if the device has iOS 10 for example this code will set "Zero" padding (cuz initial padding never set)